### PR TITLE
Refactor contest results queries

### PR DESCRIPTION
**Pull request by SvetoslavUzunov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/822

Optimized the contest results queries after adding the TotalScoreSnapshot column

### CODE REVIEW

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
+        this.participantsData = participantsData;
     }
 
     public ContestResultsViewModel GetContestResults(
             Contest contest,
             bool official,
             bool isUserAdminOrLecturer,
             bool isFullResults,
-            bool isExportResults = false)
+            int? totalResultsCount,
```
[the remark this answers is not in the document]
Comment (SvetoslavUzunov) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs:36]: Good point

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs [hunk 2]
```diff
@@ -1,44 +1,49 @@
...
         var contestActivityEntity = this.activityService
-            .GetContestActivity(contest.Map<ContestForActivityServiceModel>());
+            .GetContestActivity(contestResultsModel.Contest.Map<ContestForActivityServiceModel>());
 
         var contestResults = new ContestResultsViewModel
             {
-                Id = contest.Id,
-                Name = contest.Name,
-                IsCompete = official,
+                Id = contestResultsModel.Contest.Id,
```
Comment (shsimeonova) [inline Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs:39]: Can you do this using an Automapper configuration?

**Review by shsimeonova — APPROVED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs (hunk 2 above)
Comment (shsimeonova) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs:39]: Commenting again so i can requqst changes

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs (hunk 2 above)
Comment (SvetoslavUzunov) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs:39]: I think I will need a little help with this

Comment (SvetoslavUzunov) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs]: Update: Can you check the commit?

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs [hunk 3]
```diff
@@ -47,37 +59,62 @@ public ContestResultsAggregatorService(IParticipantsCommonDataService participan
...
+        // Get the requested participants without their problem results.
+        // Splitting the queries improves performance and avoids unexpected results from joins with Scores.
+        var participants = this.GetParticipantsPage(contest, official, page, itemsInPage)
+                .Select(ParticipantResultViewModel.FromParticipant)
+                .ToList();
+
+        // Get the ParticipantScores with another query and map problem results for each participant.
+        var participantScores = this.participantScoresDataService
+            .GetAllByParticipants(participants.Select(p => p.Id))
+            .AsNoTracking();
```
[the remark this answers is not in the document]
Comment (SvetoslavUzunov) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs]: it's fixed

File: Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs [hunk 4]
```diff
@@ -1,30 +1,42 @@
...
     }
 
     public ContestResultsViewModel GetContestResults(
             Contest contest,
             bool official,
             bool isUserAdminOrLecturer,
             bool isFullResults,
-            bool isExportResults = false)
+            bool isExportResults = false,
+            int? totalResultsCount = null,
```
[the remark this answers is not in the document]
Comment (SvetoslavUzunov) [reply Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs]: it's fixed

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**